### PR TITLE
Add start_link/1 needed by poolboy

### PR DESCRIPTION
**Pull request by zuiderkwast**
With this, we can get rid of eredis_cluster_pool_worker and use eredis directly.

This commit also adds the type host() which allows the host to be a tuple, because the code allows that (even if Dialyzer doesn't find it).

### CODE REVIEW

File: src/eredis.erl [hunk 1]
```diff
@@ -30,11 +32,24 @@
 start_link() ->
     start_link("127.0.0.1", 6379).
 
--spec start_link(Host::list(), Port::integer()) -> {ok, Pid::pid()} | {error, Reason::term()}.
+%% Poolboy needs start_link/1
+-spec start_link([option() | {host, host()} | {port, inet:port_number()}]) ->
+          {ok, pid()} | {error, Reason :: term()}.
```
Comment (bjosv) [inline src/eredis.erl:37]: space before/after ::

**Review by bjosv — APPROVED** [review]

File: src/eredis.erl (hunk 1 above)
Comment (zuiderkwast) [reply src/eredis.erl:37]: I just followed the style of surrounding code, but sure, I'll add spaces.

Comment (bjosv) [reply src/eredis.erl:37]: I meant there was extra spaces, i.e `Reason :: term()` chahe to `Reason::term()`

Comment (zuiderkwast) [reply src/eredis.erl:37]: Ah, OK

**Pull request merged**